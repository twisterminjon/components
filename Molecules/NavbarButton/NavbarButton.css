.navbarbutton {
  position: relative;
  height: var(--header-height);
  width: 85px;

  background-color: transparent;
  color: var(--brandcolor);

  cursor: pointer;

  text-transform: none;
  text-shadow: none;
  font-weight: var(--font-base--regular);
  font-size: var(--font-base-body--size-small);
  line-height: 1em;
  font-style: normal;
  text-align: center;
  text-decoration: none;
  border: none;
  outline: none;
}

.navbarbutton-content {
  display: inline-flex;
  align-items: center;
  flex-direction: column;
  flex-wrap: nowrap;
}

.navbarbutton-icon {
  position: relative;
}

.navbarbutton-badge {
  position: absolute;
  top: -10px;
  left: 20px;
}

@media only screen and (min-width: 780px) {
  .navbarbutton {
    width: 120px;
  }

  .navbarbutton-content {
    flex-direction: row;
  }

  .navbarbutton-content > span {
    margin-left: 7px;
  }

  .navbarbutton > .messagecountbubble {
    left: 32px;
  }
}

@media only screen and (max-width: 590px) {
  .navbarbutton {
    width: 60px;
  }
}

.navbarbutton-content > span {
  font-size: 12px;
  font-weight: 300;
  white-space: pre;

  margin-top: 4px;
}

.navbarbutton:hover {
  background-color: var(--brandcolor_trans_25);
}

.navbarbutton:focus {
  box-shadow: inset 0px 0px 2px 0px #000000;
}

.navbarbutton--active {
  color: var(--brandcolor_dark);
}
