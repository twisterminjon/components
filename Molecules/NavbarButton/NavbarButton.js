import React from 'react'
import { PropTypes } from 'prop-types'

import IconsNavbar from '../../Atoms/IconsNavbar/IconsNavbar'
import NotificationBadge from '../../Atoms/NotificationBadge/NotificationBadge'

import './NavbarButton.css'

NavbarButton.propTypes = {
  /** The type of button to display */
  type: PropTypes.oneOf(['home', 'staff', 'patients', 'message', 'calls']).isRequired,

  /** Can be in an active state */
  active: PropTypes.bool,

  /** Can show a notification badge with a count */
  count: PropTypes.number,

  /** Can show the count in various colors */
  countColor: PropTypes.oneOf(['red', 'yellow']),

  /** Called after the button is clicked */
  onClick: PropTypes.func.isRequired,

  /** ClassName for the wrapper */
  className: PropTypes.string,

  /** Style for the wrapper */
  style: PropTypes.object,
}
NavbarButton.defaultProps = {
  active: false,
  count: null,
  countColor: 'yellow',
  className: '',
  style: {},
}

export default function NavbarButton({ type, active, count, countColor, onClick, className, style }) {
  const types = {
    home: {
      label: 'Home',
    },
    staff: {
      label: 'Staff',
    },
    patients: {
      label: 'Patients',
    },
    message: {
      label: 'Messages',
    },
    calls: {
      label: 'Calls',
    },
  }

  const activeClass = active ? 'navbarbutton--active' : ''

  return (
    <button
      onClick={onClick}
      className={`navbarbutton ${className} ${activeClass}`.trim()}
      style={style}
      data-testid={`button-navbar-${type}`}>
      <div className={`navbarbutton-content`}>
        <div className="navbarbutton-icon">
          <IconsNavbar name={type} />
          {count > 0 && (
            <NotificationBadge count={count} className="navbarbutton-badge" negative={countColor === 'red'} />
          )}
        </div>
        <span>{types[type].label}</span>
      </div>
    </button>
  )
}
