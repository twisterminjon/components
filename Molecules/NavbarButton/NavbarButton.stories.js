import React from 'react'

import centered from '@storybook/addon-centered/react'
import { storiesOf } from '@storybook/react'
import { withKnobs, select, boolean } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import NavbarButton from './NavbarButton'

const stories = storiesOf('Provider/v1/Molecules', module)
stories.addDecorator(withKnobs)
stories.addDecorator(centered)

stories.add('NavbarButton', () => {
  const activeVal = boolean('active', false)

  const typeOpts = {
    home: 'home',
    staff: 'staff',
    patients: 'patients',
    message: 'message',
    calls: 'calls',
  }
  const typeVal = select('type', typeOpts, 'home')

  return (
    <div style={{ backgroundColor: 'white', height: '100%' }}>
      <NavbarButton type={typeVal} onClick={action('clicked')} active={activeVal} />
    </div>
  )
})
