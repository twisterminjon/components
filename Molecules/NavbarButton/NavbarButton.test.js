import React from 'react'
import NavbarButton from './NavbarButton'

describe('NavbarButton', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<NavbarButton type="home" onClick={mockFun} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls the passed function when clicked', () => {
    const wrapper = window.mount(<NavbarButton type="home" onClick={mockFun} />)

    const button = wrapper.find('[data-testid="button-navbar-home"]')
    button.simulate('click')

    expect(mockFun).toHaveBeenCalled()
  })
})
