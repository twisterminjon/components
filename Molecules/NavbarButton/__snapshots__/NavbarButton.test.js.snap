// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarButton matches the snapshot 1`] = `
<button
  className="navbarbutton"
  data-testid="button-navbar-home"
  onClick={[MockFunction]}
  style={Object {}}
>
  <div
    className="navbarbutton-content"
  >
    <div
      className="navbarbutton-icon"
    >
      <IconsNavbar
        className=""
        name="home"
        style={Object {}}
      />
    </div>
    <span>
      Home
    </span>
  </div>
</button>
`;
